lib/utils/task-transform.ts

```diff
@@ -1,6 +1,15 @@
+/**
+ * Regex to find valid checkboxes within a string
+ *
+ * Checkboxes are valid if they:
+ * - exist as the first non-whitespace on a line
+ * - take one of the following forms ( - [ ] | - [x] | - [X])
+ */
+export const checkboxRegex: RegExp = /^(\s*)- \[( |x|X)\](\s)/gm;
+
 export const withCheckboxCharacters = (s: string): string =>
   s.replace(
-    /^(\s*(?:[-+*\u2022]\s)?)- \[( |x|X)\](\s)/gm,
```

Comment:
Isn't this going to break detection of some valid checkboxes in the editor? This is also used in the note-content-editor component. I forget what the unicode one is but @dmsnell added them the last time I left them out in a PR.

Reply:
It is fixing an inconsistency with the native apps and web preview mode. Previous to this change you could have a checkbox in a list `- - [ ]`. This change makes it so those check boxes will no longer be rendered.

Reply:
Test:
- [ ]

- sd
- - [ ]

Reply:
- [ ]

Reply:
Web editing is the only place where you could have checkboxes as part of a list:
<img width="150" alt="Screen Shot 2020-11-11 at 2 56 05 PM" src="https://user-images.githubusercontent.com/6817400/98858126-19825700-242e-11eb-82cd-2cd736e591db.png">

Reply:
<img width="292" alt="Screen Shot 2020-11-11 at 3 04 57 PM" src="https://user-images.githubusercontent.com/6817400/98858882-526efb80-242f-11eb-9292-63adcdd4d675.png">

Production is also broken if you have more than one space before the checkox in a list.

Reply:
What is `\u2022`?

I'm not sure I agree we shouldn't support this.

Reply:
`\u2022` is the Unicode character for a bullet: `•`